File: labs/05/Split.java
/* Split.java is a class and program that splits a string
 * into two substrings.
 * 
 * Begun by: Serita Nelesen
 * Completed by: Zach Wibbenmeyer
 * Date: March 9, 2016
 ***************************************************************/

// import the necessary libraries
import java.util.Scanner;

// java class Split
public class Split
{
  // main program
  public static void main(String[] args)
    {
      Scanner keyboard = new Scanner(System.in);
      // prompt for the string
      System.out.println("\nTo split a string, enter the string: ");
      String theString = keyboard.next();
      
      // prompt for splitting position
      System.out.println("\nEnter the split position: ");
      int position = keyboard.nextInt();

      //Create list, compute results and print
      String [] resultList = new String[2];
      splitter(theString, position, resultList);
      System.out.println("\nThe first part is " + resultList[0] + "\n and the second part is " + resultList[1]);
      
    }
  
  /* splitter() splits a string according to the position specified
   * Receive: 	theString - the string to be split
   * 		pos - the position at which to split
   *            results - an array of strings in which to place results
   *****************************************************/
  // replace this line with a definition for splitter

  public static void splitter(String theString, int pos, String[] results) {
      
      // calculate the first part of the string
      results[0] = theString.substring(0, pos);
      // calculate the second part of the string
      results[1] = theString.substring(pos, theString.length() );
  }


}

File: labs/05/split.rb
# split.rb splits a string into two substrings.
#
# Input: A string and an integer string position to split the string by.
# Precondition: The position integer is within the range of the string size.
# Output: The substrings aString(0, position - 1) and
#   aString(pos, length() - 1).
#
# Begun by: Prof. Adams, for CS 214 at Calvin College.
# Completed by: Zach Wibbenmeyer
# Date: March 9, 2016
#################################################################


# split() splits a string into its two substrings.
# Receive: aString, the string to be split,       
#        position, the split position.                  
# PRE: 0 <= pos < the length of aString.
# Return: the substring of aString before position, and  
#         the substring of aString from position onwards.
######################################################

def split(aString, position)
   # split the string
   results = [aString[0, position], aString[position..aString.size]]
end

# main program
def main
  # prompt the user for a string
  print "To split a string, enter the string: " 
  # receive the string
  the_string = gets
  # prompt the user for a split position
  print "Enter the split position: "
  # receive the split position
  position = gets.to_i
  split_string = split(the_string, position)
  puts "The first part is: " << split_string[0]
  puts "and the second part is: " << split_string[1]
end

main
